### PR TITLE
Revise adoption of monitoring services

### PR DESCRIPTION
**Pull request by p-se**
Revise monitoring adoption section

- Add docs about how to set Node-Exporter image in cephadm for local registry.
- Be more clear about having to remove the Node-Exporter service from all nodes.
- Add a command to remove Node-Exporter from all nodes at once with Salt.
- Other text revisions.

Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1181301

Signed-off-by: Patrick Seidensal <pseidensal@suse.com>

### CODE REVIEW

File: xml/admin_ceph_upgrade.xml [hunk 1]
```diff
@@ -37,49 +39,59 @@
  </itemizedlist>
  <warning>
   <para>
-   The upgrade information in this chapter <emphasis>only</emphasis> applies to upgrades from &deepsea; to &cephadm;.
-   Do not attempt to follow these instructions if you want to deploy &productname; on the &caasp;.
+   The upgrade information in this chapter <emphasis>only</emphasis> applies to
+   upgrades from &deepsea; to &cephadm;. Do not attempt to follow these
+   instructions if you want to deploy &productname; on the &caasp;.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:44]:
```suggestion
   instructions if you want to deploy &productname; on &caasp;.
```

Comment (lproven) [reply xml/admin_ceph_upgrade.xml:44]: I think we do not usually use articles with product names like this.

File: xml/admin_ceph_upgrade.xml [hunk 2]
```diff
@@ -37,49 +39,59 @@
...
   </para>
  </warning>
  <important>
   <para>
-   Upgrading from &productname; versions older than &prevproductnumber; is not supported.
-   You first need to upgrade to the latest version of &productname; &prevproductnumber; and then follow the steps in this chapter.
+   Upgrading from &productname; versions older than &prevproductnumber; is not
+   supported. You first need to upgrade to the latest version of &productname;
+   &prevproductnumber; and then follow the steps in this chapter.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:51]:
```suggestion
   supported. First, you must upgrade to the latest version of &productname;
   &prevproductnumber;, and then follow the steps in this chapter.
```

File: xml/admin_ceph_upgrade.xml [hunk 3]
```diff
@@ -37,49 +39,59 @@
...
-    Before upgrading, ensure you read through the following sections to ensure you understand all tasks that need to be executed.
+    Before upgrading, ensure you read through the following sections to ensure
+    you understand all tasks that need to be executed.
    </para>
    <itemizedlist>
     <listitem>
      <para>
-      <emphasis>Read the release notes</emphasis> - there you can find additional information on changes since the previous release of &productname;.
-      Check the release notes to see whether:
+      <emphasis>Read the release notes</emphasis> - there you can find
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:92]:
```suggestion
      <emphasis>Read the release notes</emphasis>. In them, you can find
```

File: xml/admin_ceph_upgrade.xml [hunk 4]
```diff
@@ -99,59 +111,79 @@
...
+      The cluster upgrade may take a long time&mdash;approximately the time it
+      takes to upgrade one machine multiplied by the number of cluster nodes.
      </para>
     </listitem>
     <listitem>
      <para>
-      You need to upgrade the &smaster; first, and replace &deepsea; with &cephsalt; and &cephadm; after.
-      You will <emphasis>not</emphasis> be able to start using the &cephadm; orchestrator module until at least all &mgr;'s are upgraded.
+      You need to upgrade the &smaster; first, and replace &deepsea; with
+      &cephsalt; and &cephadm; after. You will <emphasis>not</emphasis> be able
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:145]:
```suggestion
      You need to upgrade the &smaster; first, then replace &deepsea; with
      &cephsalt; and &cephadm;. You will <emphasis>not</emphasis> be able
```

File: xml/admin_ceph_upgrade.xml [hunk 5]
```diff
@@ -99,59 +111,79 @@
...
+      to start using the &cephadm; orchestrator module until at least all
+      &mgr;'s are upgraded.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:147]:
```suggestion
      &mgr; nodes are upgraded.
```

File: xml/admin_ceph_upgrade.xml [hunk 6]
```diff
@@ -99,59 +111,79 @@
...
      </para>
     </listitem>
     <listitem>
      <para>
-      The upgrade from using &prevcephname; RPMs to &cephname; containers needs to happen all in one step.
-      This means upgrade an entire node at a time, not one daemon at a time.
+      The upgrade from using &prevcephname; RPMs to &cephname; containers needs
+      to happen all in one step. This means upgrade an entire node at a time,
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:153]:
```suggestion
      to happen in a single step. This means upgrading an entire node at a time,
```

File: xml/admin_ceph_upgrade.xml [hunk 7]
```diff
@@ -99,59 +111,79 @@
...
-         If this is going to be a problem, consider deploying these services separately on additional nodes before upgrading, so that they are down for the least amount of time possible.
-         This is the duration of the upgrade of the gateway nodes, not the duration of the upgrade of the entire cluster.
+         &mds;s and &ogw;s are down from the time the nodes are upgraded from
+         &prevcephos; to &cephos; until the services are redeployed at the end
+         of the upgrade procedure. This is particularly important to bear in
+         mind if these services are colocated with MONs, MGRs or OSDs, because
+         in this case they may be down for the entire duration of the cluster
+         upgrade. If this is going to be a problem, consider deploying these
+         services separately on additional nodes before upgrading, so that they
+         are down for the least amount of time possible. This is the duration
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:176]:
```suggestion
         are down for the shortest possible time. This is the duration
```

File: xml/admin_ceph_upgrade.xml [hunk 8]
```diff
@@ -99,59 +111,79 @@
...
+         of the upgrade of the gateway nodes, not the duration of the upgrade
+         of the entire cluster.
         </para>
        </important>
       </listitem>
       <listitem>
        <para>
-        &ganesha; and &igw;s are down only while nodes are rebooting during upgrade from &prevcephos; to &cephos; and again briefly when each service is redeployed in containerized mode.
+        &ganesha; and &igw;s are down only while nodes are rebooting during
+        upgrade from &prevcephos; to &cephos; and again briefly when each
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:185]:
```suggestion
        upgrade from &prevcephos; to &cephos;, and again briefly when each
```

File: xml/admin_ceph_upgrade.xml [hunk 9]
```diff
@@ -99,59 +111,79 @@
...
+        service is redeployed in containerized mode.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:186]:
```suggestion
        service is redeployed on the containerized mode.
```

File: xml/admin_ceph_upgrade.xml [hunk 10]
```diff
@@ -162,48 +194,69 @@
   <sect2 xml:id="upgrade-backup-config-data">
    <title>Backing Up cluster configuration and data</title>
    <para>
-    We strongly recommend backing up all cluster configuration and data before starting your upgrade to &productname; &productnumber;.
-    For instruction on how to back up all your data, see <link xlink:href="https://documentation.suse.com/ses/6/single-html/ses-admin/#cha-deployment-backup"/>.
+    We strongly recommend backing up all cluster configuration and data before
+    starting your upgrade to &productname; &productnumber;. For instruction on
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:198]:
```suggestion
    starting your upgrade to &productname; &productnumber;. For instructions on
```

File: xml/admin_ceph_upgrade.xml [hunk 11]
```diff
@@ -162,48 +194,69 @@
...
+    &productname; 5 to 6. The <option>configuration_init:
+    default-import</option> option is set in
+    <filename>/srv/pillar/ceph/proposals/config/stack/default/ceph/cluster.yml</filename>.
    </para>
    <para>
-    If <option>configuration_init</option> is still set to <option>default-import</option>, the cluster is using <filename>ceph.conf.import</filename> as its configuration file and not &deepsea;'s default <filename>ceph.conf</filename> which is compiled from files in <filename>/srv/salt/ceph/configuration/files/ceph.conf.d/</filename>.
+    If <option>configuration_init</option> is still set to
+    <option>default-import</option>, the cluster is using
+    <filename>ceph.conf.import</filename> as its configuration file and not
+    &deepsea;'s default <filename>ceph.conf</filename> which is compiled from
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:225]:
```suggestion
    &deepsea;'s default <filename>ceph.conf</filename>, which is compiled from
```

File: xml/admin_ceph_upgrade.xml [hunk 12]
```diff
@@ -215,17 +268,24 @@
   <sect2 xml:id="verify-previous-upgrade-patch-repos">
    <title>Verifying access to software repositories and container images</title>
    <para>
-    Verify that each cluster node has access to &cephos; and &productname; &productnumber; software repositories as well as registry of container images.
+    Verify that each cluster node has access to &cephos; and &productname;
+    &productnumber; software repositories as well as registry of container
+    images.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:273]:
```suggestion
    Verify that each cluster node has access to the &cephos; and &productname;
    &productnumber; software repositories, as well as the registry of container
    images.
```

File: xml/admin_ceph_upgrade.xml [hunk 13]
```diff
@@ -580,12 +677,16 @@ ID   CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
...
+     follow the steps described in
+     <xref linkend="deploy-cephadm-configure-registry"/>.
     </para>
    </step>
    <step>
     <para>
-     If you are <emphasis role="bold">not</emphasis> using container images from <literal>registry.suse.com</literal> but rather the locally configured registry, inform &ceph; which container image to use by running
+     If you are <emphasis role="bold">not</emphasis> using container images
+     from <literal>registry.suse.com</literal> but rather the locally
+     configured registry, inform &ceph; which container image to use by running
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:689]:
```suggestion
     from <literal>registry.suse.com</literal> but rather the
     locally-configured registry, inform &ceph; which container image to use by
     running
```

File: xml/admin_ceph_upgrade.xml [hunk 14]
```diff
@@ -642,22 +748,59 @@ ID   CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
...
+      local registry, not only the one for Node-Exporter. This step requires
+      you to do so for the Node-Exporter only, but it is advised than you set
+      all the monitoring container images in &cephadm; to point to the local
+      registry at this point.
+     </para>
+     <para>
+      If you do not do so, new deployments of monitoring services as well as
+      re-deployments will use the default &cephadm; configuration and you may
+      end up being unable to deploy services (in case of air-gapped
+      deployments) or with services deployed with mixed versions.
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:796]:
```suggestion
      end up being unable to deploy services (in the case of air-gapped
      deployments), or with services deployed with mixed versions.
```

File: xml/admin_ceph_upgrade.xml [hunk 15]
```diff
@@ -747,20 +896,27 @@ ID   CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
...
+    &ganesha; stores additional per-daemon configuration and exports
+    configuration in a &rados; pool. The configured &rados; pool can be found
+    on the <literal>watch_url</literal> line of the
+    <literal>RADOS_URLS</literal> block in the
+    <filename>ganesha.conf</filename> file. By default, this pool will be named
+    <literal>ganesha_config</literal>
    </para>
    <para>
-    Before attempting any migration, we strongly recommend to make a copy of the export and daemon configuration objects located in the RADOS pool To locate the configured RADOS pool, run the following command:
+    Before attempting any migration, we strongly recommend to make a copy of
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:917]:
```suggestion
    Before attempting any migration, we strongly recommend making a copy of
```

File: xml/admin_ceph_upgrade.xml [hunk 16]
```diff
@@ -789,7 +945,8 @@ drwx------ 9 root root 4.0K Sep 8 03:23 ..
 -rw-r--r-- 1 root root 350 Sep 8 03:30 export-3
 -rw-r--r-- 1 root root 358 Sep 8 03:30 export-4</screen>
    <para>
-    On a per node basis, an existing &ganesha; service needs to be stopped and then replaced with a container managed by &cephadm;.
+    On a per node basis, an existing &ganesha; service needs to be stopped and
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:948]:
```suggestion
    On a per-node basis, any existing &ganesha; service needs to be stopped and
```

File: xml/admin_ceph_upgrade.xml [hunk 17]
```diff
@@ -815,7 +976,8 @@ placement:
   namespace: ganesha
 </screen>
      <para>
-      For more information on creating a placement specification see <xref linkend="cephadm-service-and-placement-specs"/>.
+      For more information on creating a placement specification see
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:979]:
```suggestion
      For more information on creating a placement specification, see
```

File: xml/admin_ceph_upgrade.xml [hunk 18]
```diff
@@ -1095,11 +1283,16 @@ spec:
...
      <para>
-      Pools in &productname; 6 had the <option>pg_autoscale_mode</option> set to <option>warn</option> by default.
-      This resulted in a warning message in case of suboptimal number of PGs, but autoscaling did not actually happen.
-      The default in &productname; &productnumber; is the <option>pg_autoscale_mode</option> option is set to <option>on</option> for new pools and PGs will actually autoscale.
-      The upgrade process does not automatically change the <option>pg_autoscale_mode</option> of existing pools.
-      If you want to change it to <option>on</option> to get the full benefit of the autoscaler, see the instructions in <xref
+      Pools in &productname; 6 had the <option>pg_autoscale_mode</option> set
+      to <option>warn</option> by default. This resulted in a warning message
+      in case of suboptimal number of PGs, but autoscaling did not actually
+      happen. The default in &productname; &productnumber; is the
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:1289]:
```suggestion
      happen. The default in &productname; &productnumber; is that the
```

File: xml/admin_ceph_upgrade.xml [hunk 19]
```diff
@@ -1095,11 +1283,16 @@ spec:
...
+      <option>pg_autoscale_mode</option> option is set to <option>on</option>
+      for new pools and PGs will actually autoscale. The upgrade process does
```
Comment (lproven) [inline xml/admin_ceph_upgrade.xml:1291]:
```suggestion
      for new pools, and PGs will actually autoscale. The upgrade process does
```

**Review by lproven — CHANGES_REQUESTED** [review]

**Comment by p-se** [comment]
@lproven thanks for the thorough review!

**Review by tbazant — APPROVED** [review]
looks good

**Comment by tbazant** [comment]
@p-se as the PR is almost unreadable due to different formatting of the source and target file, i've reformatted the source in SUSE:main to line-break at 80 chars, mayeb you can rebase from that to make the PR more readable?

**Review by lproven — APPROVED** [review]

**Review by lproven — APPROVED** [review]

**Pull request merged**